File: backend/ReservationService/Model/Reservation.cs
using ReservationService.Attributes;
using ReservationService.Enums;

namespace ReservationService.Model
{
    [BsonCollection("reservations")]
    public class Reservation: Document
    {
        public string AccommodationId { get; set; }
        public string HostId { get; set; }
        public string GuestId { get; set; }
        public int GuestCount { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        public ReservationStatus Status { get; set; }

        public Reservation(string accommodationId, string hostId, string guestId, int guestCount, DateTime startDate, DateTime endDate, ReservationStatus status)
        {
            AccommodationId = accommodationId;
            HostId = hostId;
            GuestId = guestId;
            GuestCount = guestCount;
            StartDate = startDate;
            EndDate = endDate;
            Status = status;
        }
    }
}

File: frontend/src/app/model/reservation.ts
export class Reservation {
    id : string = ''
    hostId: string = ''
    guestId : string = ''
    accommodationId:string=''
    startDate:Date=new Date()
    endDate:Date=new Date()
    guestCount:number=0
    reservationStatus:any

    public constructor(obj? : Reservation){
        if(obj){
            this.id = obj.id
            this.hostId = obj.hostId
            this.guestId = obj.guestId
            this.accommodationId = obj.accommodationId
            this.startDate = obj.startDate
            this.endDate = obj.endDate
            this.guestCount = obj.guestCount
            this.reservationStatus = obj.reservationStatus
        }
    }
}
